lib/catalog/update_order_item.rb

```diff
@@ -1,31 +1,28 @@
 module Catalog
   class UpdateOrderItem
-    def initialize(topic, task, order_item = nil)
-      @payload    = topic.payload
-      @message    = topic.message
+    CRHC_PREFIX = 'expose_to_cloud_redhat_com_'.freeze
+
+    def initialize(task, order_item = nil)
       @task       = task
       order_item ||= OrderItem.find_by!(:topology_task_ref => @task.id)
       @order_item = order_item
     end
 
     def process
-      Rails.logger.info("Processing service order topic message: #{@message} with payload: #{@payload}")
-
-      @order_item.update_message("info", "Task update message received with payload: #{@payload}")
+      @order_item.update_message("info", "Task update message received with payload: #{@task}")
 
       mark_item_based_on_status
     end
 
     private
 
     def mark_item_based_on_status
-      case @payload["status"]
+      case @task.status
       when "ok"
-        case @payload["state"]
+        case @task.state
         when "completed"
-          @order_item.mark_completed("Order Item Complete", :service_instance_ref => service_instance_id)
+          @order_item.mark_completed("Order Item Complete", :service_instance_ref => service_instance_id, :artifacts => artifacts)
         when "running"
-          @order_item.update_message("info", "Order Item being processed with context: #{@payload["context"]}")
```

Comment:
Remove because the context is already included in the progress message at line 12